packages/app/src/cli/services/dev/processes/dev-session/dev-session.ts

```diff
@@ -77,6 +100,53 @@ export class DevSession {
     await this.logger.debug(
       `✅ Event handled [Network: ${endHRTimeInMs(networkStartTime)}ms - Total: ${endHRTimeInMs(event.startTime)}ms]`,
     )
+
+    if (this.appEventsQueue.length > 0) {
+      await this.processEvents()
+    }
+
+    this.updateInProgress = false
```

Comment:
Related to the issue on 91, probably this should be happening in a `finally` block so its unset whenever the function exits. That said, I do think tracking the promise is less prone to mistakes